src/lsp/main.ts

```diff
@@ -552,42 +602,120 @@ function registerDiagnosticsCommands(context: vscode.ExtensionContext): void {
+    }
+}
+
+async function maybeDownloadLspServer(forceDownLoad = false): Promise<string> {
     const userConfiguredClojureLspPath = config.getConfig().clojureLspPath;
-    let clojureLspPath =
-        userConfiguredClojureLspPath === ''
-            ? getClojureLspPath(extensionPath, util.isWindows)
-            : userConfiguredClojureLspPath;
-    if (userConfiguredClojureLspPath === '') {
-        const configuredVersion: string = config.getConfig().clojureLspVersion;
-        const downloadVersion = ['', 'latest'].includes(configuredVersion)
-            ? await getLatestVersion()
-            : configuredVersion;
-        if (currentVersion !== downloadVersion && downloadVersion !== '') {
-            const downloadPromise = downloadClojureLsp(
-                context.extensionPath,
-                downloadVersion
+    if (userConfiguredClojureLspPath !== '') {
+        clojureLspPath = userConfiguredClojureLspPath;
+        if (forceDownLoad) {
+            void vscode.window.showErrorMessage(
+                `Not downloading, because 'calva.clojureLSPPath' is configured (${userConfiguredClojureLspPath})`
             );
-            lspStatus.text = '$(sync~spin) Downloading clojure-lsp';
-            lspStatus.show();
-            clojureLspPath = await downloadPromise;
-            lspStatus.hide();
         }
+        return undefined;
+    } else {
+        return await ensureServerDownloaded(forceDownLoad);
     }
-    await startClient(clojureLspPath, context, handler);
 }
 
-function deactivate(): Promise<void> {
-    const client = getStateValue(LSP_CLIENT_KEY);
-    if (client) {
-        return client.stop();
+async function downloadLSPServerCommand() {
+    lspStatus.show();
+    const version = await maybeDownloadLspServer(true);
+    void vscode.window.showInformationMessage(
+        `Downloaded clojure-lsp version: ${version}`
+    );
+}
+
+async function ensureServerDownloaded(forceDownLoad = false): Promise<string> {
+    const currentVersion = readVersionFile(extensionContext.extensionPath);
+    const configuredVersion: string = config.getConfig().clojureLspVersion;
+    const downloadVersion = ['', 'latest'].includes(configuredVersion)
+        ? await getLatestVersion()
+        : configuredVersion;
+    if (
+        (currentVersion !== downloadVersion && downloadVersion !== '') ||
+        forceDownLoad
+    ) {
+        const downloadPromise = downloadClojureLsp(
+            extensionContext.extensionPath,
+            downloadVersion
+        );
+        updateStatusItem('downloading', downloadVersion);
+        clojureLspPath = await downloadPromise;
+        updateStatusItem('stopped');
+    } else {
+        clojureLspPath = getClojureLspPath(
+            extensionContext.extensionPath,
+            util.isWindows
+        );
     }
+    return readVersionFile(extensionContext.extensionPath);
+}
+
+function deactivate(): Promise<void> {
+    void stopClient();
     return Promise.resolve();
```

Comment:
I think you can just `return stopClient()` here and remove the `Promise.resolve()`.